### ISSUE TITLE
Copy-pasting from the timeline is very unreliable

### ISSUE THREAD

**Issue by KitsuneRal** [quotient-im/Quaternion#732]
One has to find out which specific combination of mouse and keyboard ever works, and that's not really usable. There's plenty of mouse areas and keyboard interception in an attempt to make it work but the result is ultimately very fragile.

**Comment by KitsuneRal** [comment]
Not something that can be done before the release, unfortunately. For the record, part of the original intent was to prevent the message entry control from losing the focus to the timeline (which is pretty much invisible and un-navigable from the focus perspective). Perhaps the current situation is at least somewhat better than that of 0.0.9.4.

**Comment by probonopd** [comment]
Ctrl-C does not work reliably for me. Actually, selecting text from multiple messages at once does not even work for me.

Quaternion v0.0.95-beta3

**Comment by KitsuneRal** [comment]
That's right; selecting the text across a few messages is quite difficult with the current design since different messages are effectively different controls. It would be nice to have a single `QTextDocument`-backed control instead (likely improves performance, eliminates problems with contiguous selection) but that means a complete rewrite of timeline visuals, effectively.

**Comment by probonopd** [comment]
fwiw, I find myself constantly fighting to copy text out of chats. All the time. So I think the rewrite would be well worth it.
